test/tests/functional/pbs_daemon_service_user.py

```diff
@@ -0,0 +1,130 @@
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import resource
+
+from tests.functional import *
+
+
+class TestDaemonServiceUser(TestFunctional):
+
+    """
+    Test suite to test running schedulers as a non-root user
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        self.orig_pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+
+    def tearDown(self):
+        self.du.set_pbs_config(self.server.shortname,
+                               confs=self.orig_pbs_conf,
+                               append=False)
+        TestFunctional.tearDown(self)
+
+    def common_test(self, binary, runas, scheduser, msg):
+        """
+        Test if running `binary` as `runas` with
+        PBS_DAEMON_SERVICE_USER set as `scheduser`
+        Check to see `msg` is in stderr
+        """
+        if scheduser:
+            self.du.set_pbs_config(
+                self.server.hostname,
+                confs={'PBS_DAEMON_SERVICE_USER': str(scheduser)}
+            )
+        else:
+            self.du.unset_pbs_config(
+                self.server.hostname,
+                confs='PBS_DAEMON_SERVICE_USER'
+            )
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        binpath = os.path.join(pbs_conf['PBS_EXEC'], 'sbin', binary)
+        ret = self.du.run_cmd(self.server.shortname,
+                              cmd=[binpath], runas=runas)
+        self.assertEquals(ret['rc'], 1)
+        self.assertIn(msg, '\n'.join(ret['err']))
+
+    def test_sched_runas_root(self):
+        """
+        Test if running sched as root with
+        PBS_DAEMON_SERVICE_USER set as another user
+        """
+        self.common_test('pbs_sched', ROOT_USER, TEST_USER,
+                         'Must be run by PBS_DAEMON_SERVICE_USER')
+
```

Comment:
You are testing a whole bunch of ways the scheduler doesn't start if it isn't the daemon user.  I don't see a test to see if it starts properly as the daemon user.

Reply:
the fact that other tests work should be proof positive that it works, right? i can add a simple test for it though

Reply:
I disagree.  The current tests will check if the scheduler will fail to start if the scheduler is not the right user.  It doesn't make sure the scheduler starts when it is the right user.  It might never start.